### PR TITLE
fix: allow not validating SBOM

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----
Should solve some error messages when people do not have this gem installed.

### CODE REVIEW

File: Library/Homebrew/sbom.rb [hunk 1]
```diff
@@ -156,7 +156,8 @@ def write(validate: true, bottling: false)
     # will no longer be valid.
     Formula.clear_cache unless spdxfile.exist?
 
-    if validate && !valid?(bottling:)
+    valid = valid?(bottling:)
+    if validate && valid.present? && !valid
```
Comment (ZhongRuoyu) [inline Library/Homebrew/sbom.rb:160]:
```suggestion
    if validate && (valid = valid?(bottling:)).present? && !valid
```

**Comment by SMillerDev** [comment]
It seems we need a `can_validate?`

File: Library/Homebrew/sbom.rb [hunk 2]
```diff
@@ -121,12 +121,12 @@ def self.fetch_schema!
     end
   end
 
-  sig { params(bottling: T::Boolean).returns(T::Boolean) }
+  sig { params(bottling: T::Boolean).returns(T.nilable(T::Boolean)) }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/sbom.rb:124]:
```suggestion
  sig { params(bottling: T::Boolean).returns(T::Boolean) }
```

File: Library/Homebrew/sbom.rb [hunk 3]
```diff
@@ -121,12 +121,12 @@ def self.fetch_schema!
...
   def valid?(bottling: false)
     unless require? "json_schemer"
       error_message = "Need json_schemer to validate SBOM, run `brew install-bundler-gems --add-groups=bottle`!"
       odie error_message if ENV["HOMEBREW_ENFORCE_SBOM"]
-      return false
+      return nil
```
Comment (MikeMcQuaid) [inline Library/Homebrew/sbom.rb:129]:
```suggestion
      return true
```

File: Library/Homebrew/sbom.rb [hunk 4]
```diff
@@ -156,7 +156,7 @@ def write(validate: true, bottling: false)
     # will no longer be valid.
     Formula.clear_cache unless spdxfile.exist?
 
-    if validate && !valid?(bottling:)
+    if validate && (valid = valid?(bottling:)).present? && !valid
```
Comment (MikeMcQuaid) [inline Library/Homebrew/sbom.rb:159]:
```suggestion
    if validate && !valid?(bottling:)
```

**Review by MikeMcQuaid — COMMENTED** [review]
Thanks for the PR. Let's just make this a boolean, like all `?` suffixed methods should be. I don't think it makes sense to use `nil` for purity's sake when it just makes the code more complex and less idiomatic.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks @SMillerDev!

**Pull request merged**